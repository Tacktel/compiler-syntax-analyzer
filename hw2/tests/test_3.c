char foo() {
    char a;

    return a;
}

char globalvar;

int main(int a, int b) {

    int a;

    a = 0;

    while (a == 2) {
        b = b + 1;
    }

    char b;

    if (a == a) {
        a = a;
    }
    else {
        b = "hello";
        b = (a + b) / 2 + 2;
    }

    while (1 == 1) {

    }

    int c;
    return 1;
}