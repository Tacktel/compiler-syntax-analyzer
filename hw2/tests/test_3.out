CHAR:char
WSPACE
ID:foo
LPAREN:(
RPAREN:)
WSPACE
LSCOPE:{
WSPACE
CHAR:char
WSPACE
ID:a
TERM:;
WSPACE
RETURN:return
WSPACE
ID:a
TERM:;
WSPACE
RSCOPE:}
WSPACE
CHAR:char
WSPACE
ID:globalvar
TERM:;
WSPACE
INT:int
WSPACE
ID:main
LPAREN:(
INT:int
WSPACE
ID:a
COMMA:,
WSPACE
INT:int
WSPACE
ID:b
RPAREN:)
WSPACE
LSCOPE:{
WSPACE
INT:int
WSPACE
ID:a
TERM:;
WSPACE
ID:a
WSPACE
ASSIGN:=
WSPACE
INTVAL:0
TERM:;
WSPACE
WHILE:while
WSPACE
LPAREN:(
ID:a
WSPACE
COMP:==
WSPACE
INTVAL:2
RPAREN:)
WSPACE
LSCOPE:{
WSPACE
ID:b
WSPACE
ASSIGN:=
WSPACE
ID:b
WSPACE
OP:+
WSPACE
INTVAL:1
TERM:;
WSPACE
RSCOPE:}
WSPACE
CHAR:char
WSPACE
ID:b
TERM:;
WSPACE
IF:if
WSPACE
LPAREN:(
ID:a
WSPACE
COMP:==
WSPACE
ID:a
RPAREN:)
WSPACE
LSCOPE:{
WSPACE
ID:a
WSPACE
ASSIGN:=
WSPACE
ID:a
TERM:;
WSPACE
RSCOPE:}
WSPACE
ELSE:else
WSPACE
LSCOPE:{
WSPACE
ID:b
WSPACE
ASSIGN:=
WSPACE
CHARVAL:"hello"
TERM:;
WSPACE
ID:b
WSPACE
ASSIGN:=
WSPACE
LPAREN:(
ID:a
WSPACE
OP:+
WSPACE
ID:b
RPAREN:)
WSPACE
OP:/
WSPACE
INTVAL:2
WSPACE
OP:+
WSPACE
INTVAL:2
TERM:;
WSPACE
RSCOPE:}
WSPACE
WHILE:while
WSPACE
LPAREN:(
INTVAL:1
WSPACE
COMP:==
WSPACE
INTVAL:1
RPAREN:)
WSPACE
LSCOPE:{
WSPACE
RSCOPE:}
WSPACE
INT:int
WSPACE
ID:c
TERM:;
WSPACE
RETURN:return
WSPACE
INTVAL:1
TERM:;
WSPACE
RSCOPE:}
